ngraph/core/src/op/if.cpp

```diff
@@ -0,0 +1,323 @@
+        }
+        else if (*then_it == *else_it)
+        {
+            new_dims.push_back(Dimension(*then_it));
+        }
+        else
+        {
+            auto dim_min = std::min((*then_it).get_min_length(), (*else_it).get_min_length());
+            auto dim_max = std::max((*then_it).get_min_length(), (*else_it).get_min_length());
+            new_dims.push_back(Dimension(dim_min, dim_max));
+        }
+    }
+
+    return PartialShape(new_dims);
+}
+
+bool op::v8::If::visit_attributes(AttributeVisitor& visitor)
+{
+    NGRAPH_OP_SCOPE(v8_If_visit_attributes);
+    m_bodies[then_body_index] =
+        std::make_shared<ngraph::Function>(OutputVector{}, ParameterVector{}, "then_branch");
+    m_bodies[else_body_index] =
+        std::make_shared<ngraph::Function>(OutputVector{}, ParameterVector{}, "else_branch");
+    visitor.on_attribute("then_body", m_bodies[then_body_index]);
+    visitor.on_attribute("else_body", m_bodies[else_body_index]);
+    visitor.on_attribute("then_inputs", m_input_descriptions[then_body_index]);
+    visitor.on_attribute("else_inputs", m_input_descriptions[else_body_index]);
+    visitor.on_attribute("then_outputs", m_output_descriptions[then_body_index]);
+    visitor.on_attribute("else_outputs", m_output_descriptions[else_body_index]);
+    return true;
+}
+
+void op::v8::If::validate_and_infer_type_body(
+    std::shared_ptr<Function> body,
+    ngraph::op::util::MultiSubgraphInputDescriptionVector& input_descriptors)
+{
+    auto layer_id_map = std::map<size_t, std::shared_ptr<op::Parameter>>();
+    for (auto param : body->get_parameters())
+    {
+        layer_id_map.insert({param->get_instance_id(), param});
+    }
+    for (const auto& input_description : input_descriptors)
+    {
+        auto index = input_description->m_input_index;
+
+        auto body_parameter = body->get_parameters().at(input_description->m_body_parameter_index);
+        auto input_partial_shape = inputs().at(index).get_source_output().get_partial_shape();
+        if (input_partial_shape.is_static())
+        {
+            auto input_shape = input_partial_shape.to_shape();
+            Shape out_shape{input_shape};
+            body_parameter->set_partial_shape(out_shape);
+        }
+        else
+        {
+            body_parameter->set_partial_shape(input_partial_shape);
+        }
+    }
+    body->validate_nodes_and_infer_types();
+}
+void op::v8::If::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v8_If_validate_and_infer_types);
+    auto cond_output = inputs().at(0).get_source_output();
+
+    auto cond_partial_shape = cond_output.get_partial_shape();
+    auto cond_rank = cond_partial_shape.rank();
+    if (cond_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this, cond_rank.get_length() < 2, "Incorrect condition");
+    }
+    if (cond_partial_shape.is_static())
+    {
+        auto cond_shape = cond_partial_shape.to_shape();
+        if (cond_rank.get_length() == 1)
```

Comment:
it should be `cond_rank.get_length() == 1` or `cond_rank.get_length() < 2` as in previous if?
do we support 0d and 1d tensors for conditional input?

Reply:
I don't understand how it was fixed, could you please provide details?
do we support 0d and 1d tensors for conditional input? do we have such a test?

Reply:
I rewrited it similar loop